sdk/cosmos/azure-cosmos/docs/StatusCodes.md

```diff
@@ -0,0 +1,61 @@
+
+The column "Expected to be transient" indicates whether this error condition is to be expected to always be seen for a short period of time only (and auto-mitigated within seconds or at least few minutes should happen in the SDK and/or the service). For error conditions that might not go away automatically the description contains additional context on what to watch out for if this error is seen for longer period of times. For errors that are supposed to be "transient" it will be necessary to file a support ticket, if the error condition exists for an extended period of time. The "Additional Info" column will also point to known issues if available and possible mitigations.
+
+## Out of scope
+This document is intentionally not going into details on how resilient applications should react  to certain error conditions (and whether/how retries are recommended). There is prescriptive guidance for developers around this located here: [Designing resilient applications with Azure Cosmos DB SDKs](https://learn.microsoft.com/azure/cosmos-db/nosql/conceptual-resilient-sdk-applications)
+
+## Status codes
+
+| Status code | Substatus code | Expected to be transient | Additional info |
+| -----------------:|-----------------------:|:-----------------------:|:----------------------|
+|200|0|No| `OK`|
+|201|0|No| `Created` - returned for createItem or upsertItem when a new document was created|
+|204|0|No| `No Content` - returned when no payload would ever be returned - like for delete operations|
+|207|0|No| `Multi-Status` - returned for bulk operations when some of the item operations have failed and others succeeded. The API allows checking status codes of item operations.|
+|304|0|No| `Not Modified` - will be returned for `ReadFeed` and `ChangeFeed` operations to indicate that there are no more changes|
+|400|\*|No| `Bad Request` - indicates that the client violated some protocol constraint. See [Bad Request TSG - trouble-shooting guide](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-bad-request) for more details.|
+|400|1001|No| `Bad Request/Partition key mismatch` - indicates that the PartitionKey defined in the point operation does not match the partition key value being extracted in the service form the document's payload based on the `PartitionKeyDefinition` of the container. See [Bad Request TSG - trouble-shooting guide](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-bad-request) for more details.|
+|400|1004|No| `Bad Request/CrossPartitionQueryNotServable` - indicates that the client attempted to execute a cross-partition query, which cannot be processed with the current SDK version. Usually this means that the query uses a query construct, which is not yet supported in the SDK version being used. Upgrading the SDK might help to address the problem. See [Bad Request TSG - trouble-shooting guide](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-bad-request) for more details.|
```

Comment:
nit: Does this currently happen in the SDK? With the existence of the QueryPlan request. I remember this used to happen on older SDKs.

Reply:
Yes - I has the same thought when Kiran raised it yesterday. But given that we probably want to generalize this anyway across SDKs I added it, what is the harm when the error never happens ?